server/routes/api/teas.js

```diff
@@ -80,11 +81,11 @@ router.get("/tea", (req, res) => {
 
 router.get("/teasList/:id", function(req, res) {
   Tea.find({ userID: req.params.id }, function(err, teas) {
-    var teaMap = [];
-    var index = 0;
-    teas.forEach(function(tea) {
-      teaMap[index] = tea;
-      index++;
+    let teaMap = { allTeas: {}, teaIDs: [] };
```

Comment:
I would still have your `teasList` endpoint return an array of teas, as that seems more RESTful.  That is, if your `/tea/:id` endpoint returns an object representing a single tea, it makes sense that the `/teas` (or /teaList in this case) would return an array of those objects.  I'd let your React app take care of storing the teas in redux as `allTeas` and `teaIDs`.  @JeffreyATW , thoughts?

Reply:
Agreed.